sound/soc/bcm/hifiberry_dacplus.c

```diff
@@ -396,6 +454,34 @@ static int snd_rpi_hifiberry_dacplus_probe(struct platform_device *pdev)
 						"hifiberry-dacplus,slave");
 		leds_off = of_property_read_bool(pdev->dev.of_node,
 						"hifiberry-dacplus,leds_off");
+		auto_mute = of_property_read_bool(pdev->dev.of_node,
+						"hifiberry-dacplus,auto_mute");
+
+		/* check for HW MUTE as defined in DT-overlay
+		   active high, therefore default to HIGH to MUTE  */
+		snd_mute_gpio =	devm_gpiod_get(&pdev->dev, "mute", GPIOD_OUT_HIGH);
```

Comment:
Do you actually want this to be devm_gpiod_get_optional? That will return NULL if nothing is found, but an error if there is some other problem.

Reply:
Ok, followed your proposal and  modified the following if(..) to get the dev_info() correct about the selected GPIO, too. Otherwise, it will run into a NULL pointer exception.